### PR TITLE
Refactoring pipeline config to update to new spec

### PR DESCRIPTION
**Pull request by sunishsheth2009**
## What changes are proposed in this pull request?
Refactoring pipeline config to update to new spec

## How is this patch tested?
Verified it here: https://github.com/mlflow/mlflow/pull/7125

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
Refactoring the template repo with the new spec for the pipeline

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/regression`: Sklearn regression pipeline example
- [ ] `area/build`: Build and test infrastructure for MLflow pipeline example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -461,31 +465,36 @@ To register models to a different server, specify the desired server in the `mod
 <details>
 <summary><strong><u>Full configuration reference</u></strong></summary>
 
-- `uri`: string. Required, if this section is present.  
+- `registry_uri`: string. Required, if this section is present.  
 URI of the model registry server to which to register trained models.
 
+- `model_name`: string. Required.
```
Comment (jinzhang21) [inline README.md:481]: On second thought, do we have to require this name? Can't it just be the basename of the project directory by default, if user doesn't specify it?

File: README.md [hunk 2]
```diff
@@ -461,31 +465,36 @@ To register models to a different server, specify the desired server in the `mod
...
+Specifies the name to use when registering the trained model to the model registry.
+
+- `model_uri`: string. Optional.  
+Specifies the URI of the model to use for batch scoring. If empty, the latest model version produced
+by the register step is used. If the register step was cleared, the latest version of the 
+registered model specified by the `model_name` attribute of the `pipeline.yaml` [register step](#register-step) 
+will be used.  
+<u>Example</u>: 
+  ```
+  model_uri: models/model.pkl
```
Comment (jinzhang21) [inline README.md]: I don't think `model_uri` should be specified here. This is an optional argument goes specifically with `predict` step.

File: README.md (hunk 1 above)
Comment (sunishsheth2009) [reply README.md:481]: Currently it is a required field. Should we try and make that optional?

File: README.md (hunk 2 above)
Comment (sunishsheth2009) [reply README.md]: Previously this was part of the predict step config and now it is not we moved all the model related config to model_registry part. So I moved the documentation for that here as well. Is that not intended? The field is still marked as optional here.

Comment (jinzhang21) [reply README.md]: @kriscon-db and @jerrylian-db for FYI.
I don't think `model_uri` should be part of registry config. Sorry I missed this part when reviewing your previous PR. `model_name` however, sounds fine to me.

Comment (sunishsheth2009) [reply README.md]: Yeah! I just assumed that model_name and model_uri config changes go together along with the registry_uri . Which is why I co-located them together.
Adding @mengxr to his thoughts on this as well.

**Review by jinzhang21 — APPROVED** [review]

**Pull request closed without merging**